src/networking/index.ts

```diff
@@ -70,28 +91,30 @@ export default class Client {
       options.body = JSON.stringify(body);
     }
 
-    return fetchWithTimeout(url, options, this.timeout).then(response => {
-      const payload = {
-        status: response.status,
-        ok: response.ok,
-        headers: response.headers,
-      };
-      return response
-        .json()
-        .then(json => {
-          return {...payload, json};
-        })
-        .catch(() => {
-          return response
-            .text()
-            .then(text => {
-              return {...payload, text};
-            })
-            .catch(() => {
-              return {...payload, text: response.statusText};
-            });
-        });
-    });
+    return fetchWithTimeout(url, options, this.timeout).then(
+      (response: any) => {
```

Comment:
If we correctly type fetchWithTimeout, we should be able to get rid of `any` here.